roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,147 @@
+        ]
+
+    @staticmethod
+    def rpm_docker_images(version):
+        return [
+            "openshift3/registry-console:{version}".format(version=version)
+        ]
+
+    @staticmethod
+    def qualified_docker_images(openshift_prefix, version):
+        return [
+            "{openshift_prefix}-haproxy-router:{version}".format(openshift_prefix=openshift_prefix, version=version),
+            "{openshift_prefix}-docker-registry:{version}".format(openshift_prefix=openshift_prefix, version=version),
+            "{openshift_prefix}-deployer:{version}".format(openshift_prefix=openshift_prefix, version=version),
+            "{openshift_prefix}-pod:{version}".format(openshift_prefix=openshift_prefix, version=version),
+        ]
+
+    def inspect_images(self, docker_path, docker_image, registry, tagged_images, tmp, task_vars):
+        failed = set()
+        for image in tagged_images:
+            # the run command pulls the {skopeo_image} if it has not already been pulled.
+            # if pulling this image fails, the current image will be added to the failed set.
+            cmd_str = "{docker_path} run {skopeo_image} skopeo inspect docker://{registry}/{tagged_image}".format(
+                docker_path=docker_path,
+                skopeo_image=docker_image,
+                registry=registry,
+                tagged_image=image
+            )
+
+            # use a "docker run ..." command on the remote host to inspect each specific
+            # image inside of a Skopeo image container. If stdout contains zero lines,
+            # it is safe to assume that an image with the specified tag does not exist
+            # in the given registry (all output in case of image not found goes to stderr).
+            cmd = self.exec_cmd(cmd_str, task_vars)
+            if not len(cmd.get("stdout_lines")):
+                failed.add(image)
+
+        return {"failed_images": failed}
+
+    def inspect_local_images(self, tagged_images, task_vars):
+        failed = set()
+        for image in tagged_images:
+            cmd = self.module_executor("docker_image_facts", {"name": image}, task_vars)
+            if not len(cmd.get("images")):
+                failed.add(image)
+
+        return {"failed_images": failed}
+
+    # ensures that the skopeo docker image exists, or can be
+    # pulled, and that the skopeo binary can be executed in it.
+    def check_skopeo_image(self, docker_path, docker_image, tmp, task_vars):
+        cmd_str = "{docker_path} run {skopeo_image} skopeo --version".format(
+                docker_path=docker_path,
+                skopeo_image=docker_image,
+        )
+
+        cmd = self.exec_cmd(cmd_str, task_vars)
+        stderr = cmd.get("stderr")
+        if len(stderr):
+            return {"failed": True, "msg": stderr}
+
+        return {}
+
+    def exec_cmd(self, cmd_str, task_vars):
+        cmd = self.module_executor("command", {"_raw_params": cmd_str}, task_vars)
+        return cmd
```

Comment:
No need for the `cmd` variable?
Better no name than a bad name ;-) it is not a cmd / command, but the result of calling one.